lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -856,6 +856,109 @@ SwiftLanguage::GetHardcodedSynthetics() {
           }
           return nullptr;
         });
+    g_formatters.push_back([](lldb_private::ValueObject &valobj,
+                              lldb::DynamicValueType dyn_type,
+                              FormatManager &format_manager)
+                               -> lldb::SyntheticChildrenSP {
+      // Try to recognize a Swift type wrapped in a C++ interop wrapper class.
+      // These types have a typedef from a char to the swift mangled name, and a
+      // static constexpr char field whose type is the typedef, and whose name
+      // is $__swift_mangled_name.
+
+
+      // This only makes sense for Clang types.
+      auto type = valobj.GetCompilerType();
+      auto tsc = type.GetTypeSystem().dyn_cast_or_null<TypeSystemClang>();
+      if (!tsc)
+        return nullptr;
+
+      ProcessSP process_sp(valobj.GetProcessSP());
+      auto *swift_runtime = SwiftLanguageRuntime::Get(process_sp);
+      if (!swift_runtime)
+        return nullptr;
+
+      auto module = valobj.GetModule();
+      if (!module) {
+        module = valobj.GetTargetSP()
+                     ->GetExecutableModule();
+        if (!module)
+          return nullptr;
+      }
+
+      auto gts = module->GetTypeSystemForLanguage(
+          lldb::LanguageType::eLanguageTypeSwift);
+      if (!gts)
+        return nullptr;
+
+      auto *ts = llvm::cast<TypeSystemSwift>(gts->get());
+
+      // We operate directly on the qualifier type because the TypeSystem
+      // interface doesn't allow us to check for static constexpr members.
+      auto qual_type = TypeSystemClang::GetQualType(type.GetOpaqueQualType());
+      auto *record_type = llvm::dyn_cast_or_null<clang::RecordType>(
+          qual_type.getTypePtrOrNull());
+      if (!record_type)
+        return nullptr;
+
+      const clang::RecordDecl *record_decl = record_type->getDecl();
+      CompilerType swift_type;
+
+      for (auto *child_decl : record_decl->decls()) {
+        auto *var_decl = llvm::dyn_cast<clang::VarDecl>(child_decl);
+        if (!var_decl)
+          continue;
+
+        auto name = var_decl->getName();
+        if (name != "$__swift_mangled_name")
+          continue;
+
+        const auto *typedef_type =
+            llvm::dyn_cast<clang::TypedefType>(var_decl->getType());
+        if (!typedef_type)
+          continue;
+
+        auto *decl = typedef_type->getDecl();
+        if (!decl)
+          continue;
+
+        auto swift_name = decl->getName();
+        if (!swift::Demangle::isMangledName(swift_name))
+          continue;
+
+        swift_type = ts->GetTypeFromMangledTypename(ConstString(swift_name));
+        break;
+      }
+
+      if (!swift_type)
+        return nullptr;
+
+      auto swift_valobj =
+          SwiftLanguageRuntime::ExtractSwiftValueObjectFromCxxWrapper(valobj);
+      if (!swift_valobj)
+        return nullptr;
```

Comment:
This code handles a bunch of different branches where it ends up not producing a synthetic backend, but you don't do any kind of logging.  You might find that you wish you had done that some day when there's a bug you can't repro locally...